velox/dwio/common/ColumnVisitors.h

```diff
@@ -1384,20 +1391,9 @@ class ExtractStringDictionaryToGenericHook {
     hook_->addNull(rowIndex);
   }
 
-  void addValue(vector_size_t rowIndex, int32_t value) {
-    // We take the string from the stripe or stride dictionary
-    // according to the index. Stride dictionary indices are offset up
-    // by the stripe dict size.
-    if (value < dictionarySize()) {
-      auto view = folly::StringPiece(
-          reinterpret_cast<const StringView*>(state_.dictionary.values)[value]);
-      hook_->addValue(rowIndex, &view);
-    } else {
-      VELOX_DCHECK(state_.inDictionary);
-      auto view = folly::StringPiece(reinterpret_cast<const StringView*>(
-          state_.dictionary2.values)[value - dictionarySize()]);
-      hook_->addValue(rowIndex, &view);
-    }
+  template <typename V>
```

Comment:
You can restore this back once we move `hasDictionary_` as a template parameter

Reply:
I tried to roll back this code to the previous logic, but an exception occurred during compilation, so I think more work needs to be done.
```
/data/home/wypb/data/code/apache/velox/./velox/dwio/common/ColumnVisitors.h:517:57: error: cannot convert ‘folly::Range<const char*>’ to ‘int32_t’ {aka ‘int’}
  517 |     values_.addValue(rowIndex_ + reader_->numScanned(), value);
      |                                                         ^~~~~
      |                                                         |
      |                                                         folly::Range<const char*>
/data/home/wypb/data/code/apache/velox/./velox/dwio/common/ColumnVisitors.h:1395:15: note:   initializing argument 2 of ‘void facebook::velox::dwio::common::ExtractStringDictionaryToGenericHook::addValue(facebook::velox::vector_size_t, int32_t)’
 1395 |       int32_t value) {
      |       ~~~~~~~~^~~~~
/data/home/wypb/data/code/apache/velox/./velox/dwio/common/ColumnVisitors.h:519:33: error: cannot convert ‘folly::Range<const char*>’ to ‘int32_t’ {aka ‘int’}
  519 |     values_.addValue(rowIndex_, value);
      |                                 ^~~~~
      |                                 |
      |                                 folly::Range<const char*>
```